dali/operators/image/crop/random_crop_attr.h

```diff
@@ -53,24 +53,36 @@ class RandomCropAttr {
     std::vector<int> seeds(max_batch_size);
     seq.generate(seeds.begin(), seeds.end());
 
-    crop_window_generators_.resize(max_batch_size);
-
+    random_crop_generators_.reserve(max_batch_size);
     for (int i = 0; i < max_batch_size; i++) {
-      std::shared_ptr<RandomCropGenerator> random_crop_generator(
+      random_crop_generators_.emplace_back(
         new RandomCropGenerator(
           {aspect_ratio[0], aspect_ratio[1]}, {area[0], area[1]}, seeds[i], num_attempts));
```

Comment:
It's generally a bad practice to `emplace_back` a dynamically allocated pointer to a collection of smart pointers, because if the allocation of a new collection element fails (e.g. due to `std::bad_alloc` being thrown), the pointer is leaked.

Safer and more efficient:
```suggestion
      random_crop_generators_.push_back(
        std::make_shared<RandomCropGenerator>({aspect_ratio[0], aspect_ratio[1]}, {area[0], area[1]}, seeds[i], num_attempts));
```